app/Providers/AppServiceProvider.php

```diff
@@ -61,6 +61,10 @@ public function register(): void
         $this->app->singleton('sensor-discovery', function (Application $app) {
             return new \App\Discovery\Sensor($app->make('device-cache')->getPrimary());
         });
+
+        $this->app->singleton(\App\Services\WidgetRegistry::class, function ($app) {
```

Comment:
Instead of registering a singleton, just register and tag all the widgets individually.